jina/peapods/runtimes/container/__init__.py

```diff
@@ -12,13 +13,30 @@
 from ....excepts import BadImageNameError, DockerVersionError
 from ...zmq import send_ctrl_message
 from ....helper import ArgNamespace, slugify
+from ....enums import SocketType
+
+if False:
+    import multiprocessing
+    import threading
+    from jina.logging.logger import JinaLogger
 
 
 class ContainerRuntime(ZMQRuntime):
     """Runtime procedure for container."""
 
-    def __init__(self, args: 'argparse.Namespace', ctrl_addr: str, **kwargs):
-        super().__init__(args, ctrl_addr, **kwargs)
+    def __init__(self, args: 'argparse.Namespace', **kwargs):
+        super().__init__(args, **kwargs)
+        self.ctrl_addr = self.get_control_address(
+            args.host,
+            args.port_ctrl,
+            docker_kwargs=self.args.docker_kwargs,
+        )
```

Comment:
Why is this different from the `_zed_runtime_ctrl_address` above?

Reply:
an alternative would be to add another argument to the parser, but I do not want to overkill with args